### ISSUE TITLE
feature request: Multiple Sign-in Experiences for Different Applications

### ISSUE THREAD

**Issue by jafarijason** [logto-io/logto#5723]
I hope this message finds you well. I wanted to take a moment to express my appreciation for the outstanding product you've developed. LogTo has truly transformed the way we handle logging processes, and we're incredibly grateful for the value it brings to our organization.

I have another feature request that I believe would further elevate the versatility and customization options within the platform. It would be incredibly beneficial to introduce functionality for creating multiple sign-in experiences, allowing administrators to tailor the authentication process to the unique needs of different applications.

This feature could be implemented through an intuitive interface in the admin console, where administrators can define and configure distinct sign-in experiences for each application within the LogTo platform. These sign-in experiences could encompass a range of authentication methods, customization options, and branding elements, providing a seamless and personalized user journey for each application.

By offering multiple sign-in experiences, LogTo would empower administrators with greater flexibility and control over the authentication process, enabling them to optimize security, user experience, and branding consistency across various applications.

I believe this feature would be a significant enhancement to the LogTo platform and would greatly appreciate your consideration in implementing it. Thank you for your ongoing commitment to innovation and customer satisfaction.

**Comment by charIeszhao** [comment]
Thanks, this is so called "per app sign-in-experience" internally, and it's already on our roadmap. Please stay tuned to our future updates.

**Comment by jafarijason** [comment]
Perfect, please also keep this issue inform once you add "per app sign-in-experience"

**Comment by maurice-freitag** [comment]
We have a soft requirement of customizing the sign-in experience depending on parameters provided by our client. One example would be a user trying to login to a specific tenant (tenant1.myapp.io). Ideally we would at least be able to display the company's logo. Even better would be to limit the available sign-in options and only allow SSO login with email addresses provided by this specific company.

Would this be covered by the "per app sign-in-experience"? I could imagine using (LogTo-)organizations to control this behavior - configure the organization to only allow specific sign-in methods, add metadata that would be displayed in the UI, etc. The client could then add the organization's identifier as a parameter to the sign-in site.

**Comment by jafarijason** [comment]
Exact @maurice-freitag, I have the same problem, @charIeszhao please consider it, thanks

**Comment by charIeszhao** [comment]
Hi, this is supported right now, using the "Bring your UI" feature. In the custom sign-in UI, you can fetch the "applicationId" from the URL query params.

https://docs.logto.io/customization/bring-your-ui

Let us know if you have further questions.

**Comment by DaleWebb** [comment]
@charIeszhao  thanks for the update.

Unless I've misunderstood something, bringing your own UI wouldn't solve the original request of limiting sign-in options per app, would it?

For example, you could change the UI to only show some options, but the Logto product would still allow any enabled sign-in option to any Logto Application.

**Comment by charIeszhao** [comment]
Yes, you have a good point. The Custom UI is front-end only, and thus it won't prevent users from hacking into your app in a sign-in option that you just hide from your page. However, doing so is extremely cumbersome and requires hacker-level skills in order to reverse engineer the sign-in page.

However, you can still open up a new feature request on our [public roadmap](https://logto.productlane.com/roadmap) if you prefer having it on the service level.

**Comment by DaleWebb** [comment]
Thanks! I found [something similar in the backlog](https://logto.productlane.com/roadmap?id=d0e6c480-8a31-4d97-91ea-e6d576688405), so I've upvoted it.

I wonder if it's possible to prevent against the hacking-the-signin-page scenario by using the PostRegister webhook to detect the signup method and then delete the newly-created user if they've signed up with a disallowed method

**Comment by charIeszhao** [comment]
Hmmm, I think it's a good idea. That'll work well for your case.
